### ISSUE TITLE
Apply style to page will mess with the footnote style 

### ISSUE THREAD

**Issue by actforjason** [stevencohn/OneMore#692]
Before
![image](https://user-images.githubusercontent.com/57058325/197403570-90fa17b0-d88c-41e6-bcb0-1220fa7890a9.png)
After
![image](https://user-images.githubusercontent.com/57058325/197403581-a8b92f42-0dbc-434e-81c5-5cf1b67eeb14.png)

**Comment by actforjason** [comment]
The superscript and color of the footnote change after the page style is applied.
Sorry, I'm a little OCD.

![image](https://user-images.githubusercontent.com/57058325/197510687-b6ebc736-03d8-47f5-9bce-e2f9d4e65be7.png)
![image](https://user-images.githubusercontent.com/57058325/197510694-ee375025-53d2-45c0-98a2-e688aaaf4152.png)

**Comment by actforjason** [comment]
There still be a little problem based on the latest commit #695. The footnote num and "刷新"(refresh) are blue, while after applying the style they become text color. I tried to modify the source code but failed.

![image](https://user-images.githubusercontent.com/57058325/199743816-b1a51182-24e1-48ed-8087-af4dded13a3e.png)
![image](https://user-images.githubusercontent.com/57058325/199743827-5549bdb2-1935-44df-b105-938f0c1182e6.png)

**Comment by stevencohn** [comment]
That cannot be fixed. See the [comment here](https://github.com/stevencohn/OneMore/blob/main/OneMore/Commands/Styles/ApplyStylesCommand.cs#L268). 

OneMore is not applying color to the hyperlink but OneNote is and I can't seem to prevent that.

**Comment by actforjason** [comment]
I don't know much about this aspect, but I think it should work if put `ApplyToHyperlinks()` after OneNote Update `onenote.UpdatePageContent()`,  currently `ApplyToHyperlinks()` seems to be before `onenote.UpdatePageContent()`.
https://github.com/stevencohn/OneMore/blob/902cf770a9d3b7c435440a0cc254aea7222c043b/OneMore/Commands/Styles/ApplyStylesCommand.cs#L72-L76

https://github.com/stevencohn/OneMore/blob/902cf770a9d3b7c435440a0cc254aea7222c043b/OneMore/OneNote.cs#L1008-L1011

**Comment by stevencohn** [comment]
ApplyToHyperlinks and most of the rest of that code just update an in-memory copy of the page XML. It's UpdatePageContent that actually sends that XML back to OneNote to be saved. So it must be the last function call.

**Comment by actforjason** [comment]
Why can change the color of text through OneNote itself, page doesn't seem to reload, and it does work in its XML.

**Comment by stevenmcohn** [comment]
It's just the way Microsoft implement it. The OneNote UI seems to update the page differently than the Interop API that they've published for plugins like OneMore. It's all OneMore has available to use.

**Comment by stevencohn** [comment]
It's just the way Microsoft implement it. The OneNote UI seems to update the page differently than the Interop API that they've published for plugins like OneMore. It's all OneMore has available to use.

**Comment by actforjason** [comment]
There seems to be a way here.
[Change OneNote hyperlink color programmatically](https://answers.microsoft.com/en-us/msoffice/forum/all/change-onenote-hyperlink-color-programmatically/d673f8f8-dcd7-4910-9403-8b45379917a6)

> The next workaround works:
> 1. Add QuickStyleDef to the page with necessary font color. 
> 2. Set the "quickStyleIndex" attribute for hyperlink's parent element  (one:OE) to the value of QuickStyleDef.Attribute("index") that we have added to the page.
> 3. Wrap all other text in that one:OE element with <span style='color:black'>{0}</span>
> 4. Insert the empty one:T element before the hyperlink 
> And the hyperlink text color will be like we defined it in QuickStyleDef.

**Comment by stevencohn** [comment]
No, quickstyles should be preserved for OneNote built-in styles and the OneMore custom styles. You're implying fabricating a quickstyle just to prevent a hyperlink from changing color but that quickstyle wouldn't be related to any style that the user could change via the UI. This is not a good solution.

Ostensibly, this is working as intended. It doesn't look 100% correct, but it's doing what it should do - at least as far as the Interop API allows. There's nothing left to do here.
